src/Symfony/Component/Messenger/TraceableMessageBus.php

```diff
@@ -41,7 +41,9 @@ public function dispatch(object $message, array $stamps = []): Envelope
 
             throw $e;
         } finally {
-            $this->dispatchedMessages[] = $context + ['stamps_after_dispatch' => array_merge([], ...array_values($envelope->all()))];
+            if ($envelope instanceof Envelope) {
```

Comment:
Can this check be removed now that it is always an Envelope? Was this left over from when it was array or Envelope?